src/data/Path.cpp

```diff
@@ -175,48 +152,50 @@ Path::Parts::SetTo(const StringPart& path)
  * \param[out] buffer   Buffer where path will be inserted.
  * \param[in]  behavior Compatibility behavior to emulate.
  */
-void
-Path::Parts::GetPath(StringBuffer& buffer, const behavior::Behavior& behavior)
-	const
+std::string
+Path::Parts::ToPath(const behavior::Behavior& behavior) const
 {
+	std::string buffer;
 	// TODO: This is platform dependent!
 
-	if (!fGrist.IsEmpty()) {
-		if (fGrist.Start()[0] != '<')
+	if (!fGrist.empty()) {
+		if (fGrist.front() != '<')
 			buffer += '<';
 		buffer += fGrist;
-		if (fGrist.End()[-1] != '>')
+		if (fGrist.back() != '>')
 			buffer += '>';
 	}
 
 	// Use root only, if the directory part isn't absolute and if the root isn't
 	// ".".
-	if (!fRoot.IsEmpty() && (fRoot.Length() > 1 || fRoot.Start()[0] != '.')
+	if (!fRoot.empty() && (fRoot.length() > 1 || fRoot.front() != '.')
 		&& !Path::IsAbsolute(fDirectory)) {
 		buffer += fRoot;
 		if (behavior.GetPathRootReplacerSlash()
 				== behavior::Behavior::PATH_ROOT_REPLACER_SLASH_ALWAYS
-			|| fRoot.End()[-1] != '/') {
+			|| fRoot.back() != '/') {
 			buffer += '/';
 		}
 	}
 
-	if (!fDirectory.IsEmpty()) {
+	if (!fDirectory.empty()) {
 		buffer += fDirectory;
-		if (fDirectory != StringPart("/", 1)
-			&& (!fBaseName.IsEmpty() || !fSuffix.IsEmpty())) {
+		if (fDirectory.back() != '/'
```

Comment:
That's correct, but the original behavior seemed strange to me as I was changing this. Normally it would add a slash even if the directory already had a trailing slash, like this:
```
D=directory/
-
directory//foo
directory//bar.a
directory//foobar.bee
directory//foo
directory//foo.a
directory//bar.bee
directory//foobar.so
directory/
<grist>directory//bar.bee(archive.o)
```

I'll revert the behavioral change and leave this to a different PR